### PR TITLE
Fix  UTs for qnx7

### PR DESCRIPTION
**Pull request by BSolonenko**
This PR is **ready** for review.  
This PR includes implementation for  proposal [Porting SDL to QNX700 and Automotive Grade Linux (AGL) x64 platforms](https://github.com/LuxoftAKutsan/sdl_evolution/blob/port_sdl_on_qnx700_and_agl/proposals/nnnn-sdl_on_qnx700_agl.md)

### Summary
Unit tests have been fixed for: 

- smart_object_test
- data_resumption_test
- generator_test
- config_profile_test
- app_launch_data_test
- application_manager_mock_hmi_test

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/utils/include/utils/sqlite_wrapper/sql/sql_database.h [hunk 1]
```diff
@@ -49,7 +49,7 @@ enum class StorageType {IN_MEMORY};
  */
 class SQLDatabase {
  public:
-  SQLDatabase(StorageType storage_type);
+  explicit SQLDatabase(Persistent);
```
Comment (LuxoftAKutsan) [inline src/components/utils/include/utils/sqlite_wrapper/sql/sql_database.h]: @BSolonenko Persistent again? Looks like type

File: src/components/utils/test/qdb_wrapper/qdbserver.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# This script star QDB server for SDL
```
Comment (LuxoftAKutsan) [inline src/components/utils/test/qdb_wrapper/qdbserver.sh]: @BSolonenko please specify in file name that it starts qdb server for tests

**Review by LuxoftAKutsan — APPROVED** [review]
@BSolonenko please take care about Persistance

File: src/components/application_manager/src/resumption/resumption_data_db.cc [hunk 3]
```diff
@@ -130,12 +130,10 @@ bool ResumptionDataDB::Init() {
     return false;
   }
   if (0 == query_checks_resumption.GetInteger(0)) {
-    utils::dbms::SQLQuery query_insert_resumption(db());
-    if (!query_insert_resumption.Prepare(kInsertInitData) ||
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/resumption/resumption_data_db.cc:134]: @BSolonenko why `prepare` was removed?

File: src/components/config_profile/test/CMakeLists.txt [hunk 4]
```diff
@@ -42,10 +42,17 @@ set(LIBRARIES
...
-file(COPY smartDeviceLink_invalid_boolean.ini DESTINATION ${CMAKE_CURRENT_BINARY_DIR}) 
+file(COPY smartDeviceLink_invalid_int.ini DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+file(COPY smartDeviceLink_invalid_boolean.ini DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+
+file(COPY smartDeviceLink.ini DESTINATION ${TEST_BIN_DIR})
+file(COPY smartDeviceLink_test.ini DESTINATION ${TEST_BIN_DIR})
+file(COPY smartDeviceLink_invalid_pairs.ini DESTINATION ${TEST_BIN_DIR})
+file(COPY smartDeviceLink_invalid_string.ini DESTINATION ${TEST_BIN_DIR})
+file(COPY smartDeviceLink_invalid_int.ini DESTINATION ${TEST_BIN_DIR})
+file(COPY smartDeviceLink_invalid_boolean.ini DESTINATION ${TEST_BIN_DIR})
```
Comment (AKalinich-Luxoft) [inline src/components/config_profile/test/CMakeLists.txt]: @BSolonenko please add an empty line in the end of file

**Review by AKalinich-Luxoft — CHANGES_REQUESTED** [review]

File: src/components/application_manager/src/resumption/resumption_data_db.cc (hunk 3 above)
Comment (BSolonenko) [reply src/components/application_manager/src/resumption/resumption_data_db.cc:134]: @AKalinich-Luxoft  `bool Exec(const std::string& query);` function combines the formatting string in format with the values of the additional arguments to construct an SQL command string and then executes that string on the database referred to by db. 

This is done as in: https://github.com/Ford-Luxoft/sdl_core/blob/d85e7f2e5a844145149a480f8e5d2d8cdcc1a013/src/components/policy/policy_external/src/sql_pt_representation.cc#L440

File: src/components/config_profile/test/CMakeLists.txt (hunk 4 above)
Comment (BSolonenko) [reply src/components/config_profile/test/CMakeLists.txt]: @AKalinich-Luxoft fixed in https://github.com/Ford-Luxoft/sdl_core/pull/21/commits/72faa1e1b1feaf2c018815063d6e5fe771cfe13e

**Review by AKalinich-Luxoft — APPROVED** [review]

**Review by mkorniichuk — APPROVED** [review]

**Review by SNiukalov — APPROVED** [review]

**Pull request merged**